src/version_set.h

```diff
@@ -86,6 +81,11 @@ class VersionSet {
 
   Status Apply(VersionEdit* edit);
 
+  // REQUIRES: write lock is held
+  Status LogAndApplyLocked(VersionEdit* edit);
+
+  port::RWMutex mutex_;
```

Comment:
I prefer mutex instead of rwlock (i.e. RWMutex) unless we have some benchmark here. My understanding is rwlock is good only for when you want to hold (read) lock for relatively longer time and rwlock overhead can be ignore. My experience with rwlock is it rarely outperform mutex in most of the cases.

Reply:
noted